services/ui-src/src/components/fields/DynamicField.tsx

```diff
@@ -1,76 +1,103 @@
 import { Box, Button, Flex, Image } from "@chakra-ui/react";
+import { TextField as CmsdsTextField } from "@cmsgov/design-system";
 import { svgFilters } from "styles/theme";
+// utils
+import { EntityShape, InputChangeEvent } from "types";
 // assets
 import cancelIcon from "assets/icons/icon_cancel_x_circle.png";
-import { TextField } from "./TextField";
 
 export const DynamicField = ({ name, label, ...props }: Props) => {
   // get form context and register field
   const form = useFormContext();
   form.register(name);
+  const [displayValues, setDisplayValues] = useState<EntityShape[]>([]);
 
   // make formfield dynamic array with config options
-  const { fields, append, remove } = useFieldArray({
+  const { append, remove } = useFieldArray({
     name: name,
     shouldUnregister: true,
   });
 
-  // render form field values as individual inputs
+  // update display value and form field data on change
+  const onChangeHandler = async (event: InputChangeEvent) => {
+    const { id, value } = event.target;
+    const currentEntity = displayValues.find((entity) => entity.id === id);
+    const currentEntityIndex = displayValues.indexOf(currentEntity!);
+    const newDisplayValues = [...displayValues];
+    newDisplayValues[currentEntityIndex].name = value;
+    setDisplayValues(newDisplayValues);
+  };
+
+  const appendNewRecord = () => {
+    const newEntity = { id: Date.now().toString(), name: "" };
+    append(newEntity);
+    setDisplayValues([...displayValues, newEntity]);
+  };
+
+  const removeRecord = (index: number) => {
+    remove(index);
+    const newDisplayValues = [...displayValues];
+    newDisplayValues.splice(index, 1);
+    setDisplayValues(newDisplayValues);
+  };
+
+  // set initial display value to form state field value or hydration value
+  const hydrationValue = props?.hydrate;
   useEffect(() => {
-    if (fields.length === 0) {
-      append(props?.hydrate || "");
+    if (hydrationValue) {
+      setDisplayValues(hydrationValue);
+      append(hydrationValue);
+    } else {
+      appendNewRecord();
     }
-  }, []);
+  }, [props?.hydrate]); // only runs on hydrationValue fetch/update
 
-  // render hydrated values on refresh
+  // on displayValue change, set field array value to match
   useEffect(() => {
-    form.reset();
-    append(props?.hydrate || "");
-  }, [props?.hydrate]);
+    form.setValue(name, displayValues, { shouldValidate: true });
+  }, [displayValues]);
 
   const fieldErrorState = form?.formState?.errors?.[name];
 
   return (
     <Box>
-      {fields.map((field: Record<"id", string>, index: number) => {
+      {displayValues.map((field: EntityShape, index: number) => {
         return (
-          <Flex key={field.id} alignItems="flex-end">
-            <TextField
+          <Flex key={field.id} sx={sx.textField}>
+            <CmsdsTextField
+              id={field.id}
               name={`${name}[${index}]`}
-              label={label}
+              label={label || ""}
```

Comment:
Don't necessarily need the "" here because it's a required prop.
```suggestion
              label={label}
```